src/main/Extensions/VSCode/VSCodeExtension.ts

```diff
@@ -0,0 +1,163 @@
+            new Database(databasePath, {})
+                .prepare(
+                    "SELECT json_extract(value, '$.entries') as entries FROM ItemTable WHERE key = 'history.recentlyOpenedPathsList'",
+                )
+                .pluck()
+                .get() as string,
+        );
+    }
+
+    async getSearchItem(recent: VscodeRecent): Promise<SearchResultItem> {
+        const uri = recent.fileUri ?? recent.folderUri;
+        const isFile = !!recent.fileUri;
+        const description = isFile ? "File" : "Folder";
+
+        let img: Image;
+        try {
+            img = await this.fileImageGenerator.getImage(uri);
+        } catch (e) {
+            img = this.getImage();
+        }
+
+        const path = URL.fileURLToPath(uri);
+
+        return {
+            id: "vscode-" + path,
+            name: Path.basename(path),
+            description,
+            image: img,
+            defaultAction: {
+                handlerId: "VSCodeHandler",
+                description: `Open ${description} in VSCode`,
+                argument: path,
+            },
+        };
+    }
+
+    public isSupported(): boolean {
+        return ["macOS", "Linux", "Windows"].includes(this.operatingSystem);
+    }
+
+    public getSettingDefaultValue<T>(key: string) {
+        const defaultSettings = {
+            prefix: "vscode",
+        };
+
+        return defaultSettings[key] as T;
+    }
+
+    public getImage(): Image {
+        const path = this.assetPathResolver.getExtensionAssetPath("VSCode", "vscode.png");
+
+        return {
+            url: `file://${path}`,
+        };
+    }
+
+    public getI18nResources() {
+        return {
+            "en-US": {
+                extensionName: "Visual Studio Code",
+                prefix: "Prefix",
+                prefixDescription:
+                    "The prefix to trigger visual studio code. Open recently opened files and projects: <prefix> <command>",
+            },
+        };
+    }
+
+    public getInstantSearchResultItems(searchTerm: string): SearchResultItem[] {
+        if (this.getPrefix().trim() !== "" && !searchTerm.startsWith(this.getPrefix() + " ")) {
+            return [];
+        }
+
+        searchTerm = searchTerm.replace(this.getPrefix() + " ", "").trim();
+
+        if (searchTerm && searchTerm === "") {
+            return this.recents;
+        }
+
+        const fuzziness = this.settingsManager.getValue<number>("searchEngine.fuzziness", 0.5);
+        const maxSearchResultItems = this.settingsManager.getValue<number>("searchEngine.maxResultLength", 50);
+        const searchEngineId = this.settingsManager.getValue<SearchEngineId>("searchEngine.id", "Fuse.js");
+
+        return searchFilter(
+            {
+                searchResultItems: this.recents,
+                searchTerm: searchTerm,
+                fuzziness: fuzziness,
+                maxSearchResultItems: maxSearchResultItems,
+            },
+            searchEngineId,
+        );
```

Comment:
Quick tip: if the variable name is the same as the name of the property that you're assigning it to, you can simplify the code:

```suggestion
        return searchFilter(
            {
                searchResultItems: this.recents,
                searchTerm,
                fuzziness,
                maxSearchResultItems,
            },
            searchEngineId,
        );
```

Reply:
These are personal preference styling things if there is something you want for the project best just ignore them in a pr and add an eslint rule after every js project has different preferences its best just ignore them and pop the in after I personally prefer the explicitness of the key value for example but if eslint put a sqwiggle under my code id have just auto fixed it. PS added the eslint rule for this 😃

Reply:
I'm not saying that you *should* do this, it was just a tip, in case you didn't know it ;) The eslint rule is not necessary IMO